# Q&A Activity on Testing of Hypothesis
1. Carry out descriptive analytics on the dataset.
2. Check whether female and male students scored the same marks.
3. Whether test preparation helps the students?

In [66]:
#importing libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import scipy.stats as stats

In [67]:
df=pd.read_csv('Week_8_Q&A_dataset - Sheet1.csv')

In [68]:
df.head()

Gender Test preparation  Total Marks
0    male             none           14
1  female             none           28
2  female             none           18
3  female             none           48
4  female             none           21

In [69]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 28 entries, 0 to 27
Data columns (total 3 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   Gender            28 non-null     object
 1   Test preparation  28 non-null     object
 2   Total Marks       28 non-null     int64 
dtypes: int64(1), object(2)
memory usage: 800.0+ bytes


# 1. Carry out descriptive analytics on the dataset.

In [70]:
df.describe()

Total Marks
count    28.000000
mean     32.321429
std      12.454520
min      12.000000
25%      22.500000
50%      33.000000
75%      43.250000
max      50.000000

# Insights:
    1.The given dataset contains 28 rows and 3 columns.
    2.This dataset is not containing any null values.
    3The average marks of the dataset is 32.32 with a standard deviation of 12.45.
    Maximum and Minimum marks are 50 and 12 respectively.

# 2. Check whether female and male students scored the same marks.
# Two Tailed two sample t-test
# Hypothesis:
*  **Ho**:The average score of female and male students is equal
*  **Ha**:The average scores are not equal
*  $\alpha $: 0.05

Performing two-tailed t-test on the dataset to analyse our hypothesis

In [71]:
df1=df.groupby('Gender')
male = df.query('Gender == "male"')['Total Marks']
female = df.query('Gender == "female"')['Total Marks']

In [72]:
df1.describe()

Total Marks                                                      
             count       mean        std   min    25%   50%    75%   max
Gender                                                                  
female        16.0  33.750000  12.865976  12.0  25.50  39.0  43.50  50.0
male          12.0  30.416667  12.168950  14.0  21.75  29.0  43.25  47.0

In [73]:
df1.head()

Gender Test preparation  Total Marks
0     male             none           14
1   female             none           28
2   female             none           18
3   female             none           48
4   female             none           21
5   female        completed           40
6     male             none           30
8     male             none           18
9     male             none           24
10    male        completed           44

In [74]:
tvalue, pvalue=stats.ttest_ind(male,female,alternative='two-sided')
print("The t value of means is: ", round(tvalue, 4))
print("The p value based on the t score is: ", round(pvalue,4))

The t value of means is:  -0.6941
The p value based on the t score is:  0.4938


## As the p-value=0.4906112127110316 > $\alpha$=0.05 and hence we cannot reject the null Hypothesis
### ie. Female and Male students scored the same marks

# 3. Whether test preparation helps the students? 
# One tailed Two sample t-test

# Hypothesis:
*  **Ho** :The average score of students who completed vs students who didn't is equal

*  **Ha** :The average scores of students who completed are higher
*  $\alpha$ : 0.05

Performing one-tailed t-test on the dataset to analyse our hypothesis

In [75]:
df2=df.groupby('Test preparation')
completed=df[df['Test preparation'] =="completed"]['Total Marks']
none=df[df['Test preparation'] =="none"]['Total Marks']

In [76]:
df2.describe()

Total Marks                                                \
                       count       mean        std   min   25%   50%   75%   
Test preparation                                                             
completed                7.0  44.571429   3.359422  40.0  43.0  44.0  46.0   
none                    21.0  28.238095  11.661495  12.0  18.0  27.0  38.0   

                        
                   max  
Test preparation        
completed         50.0  
none              48.0

In [77]:
tvalue, pvalue=stats.ttest_ind(completed,none,alternative='greater')
print("The t value of means is: ", round(tvalue, 4))
print("The p value based on the t score is: ", round(pvalue,4))


The t value of means is:  3.6144
The p value based on the t score is:  0.0006


## As the p-value 0.0006 < $\alpha$ = 0.05, we reject the null hypothesis. 
### ie. We can say that Average marks for students completed the test is higher .